lib/game_state.rb

```diff
@@ -0,0 +1,28 @@
+require_relative './token'
+require_relative '../lib/player'
+
+class GameState 
+    include Token
+    include PlayerType
+    attr_accessor :player_1, :player_2, :board, :current_player
+    
+    def initialize(
+        player_1: human,
+        player_2:,
+        board_size:
+    )
+        @board = Board.new(size: board_size)
+        @player_1 = Player.new(player: get_player(player_1).new(), token: hero)
+        @player_2 = Player.new(player: get_player(player_2).new(), token: opponent)
```

Comment:
is it weird that Game state knows how `Board` and `Player` should be initialized ? 
is this too dependent?

Reply:
I think it's fine